syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package TransportBuffers;

message Instruction {
  optional uint32 protocol_version = 1;

  optional uint64 old_num = 2;
  optional uint64 new_num = 3;
  optional uint64 ack_num = 4;
  optional uint64 throwaway_num = 5;

  optional bytes diff = 6;

  optional bytes chaff = 7;

  optional bytes oob = 8;
}
